chainerx_cc/chainerx/indexer.h

```diff
@@ -37,16 +37,16 @@ class NdimIndex {
     int8_t ndim_;
 };
 
-template <int8_t kNdim = kDynamicNdim>
+template <int8_t kNdim = kDynamicNdim, class T = IndexIterator<kNdim>>
```

Comment:
I'm not confident that it is possible, but can we specify the index iterator with a member type?
```cuda
#ifdef __CUDA_ARCH__
using indexer_type = CudaIndexIterator<kNdim>;
#else
using indexer_type = IndexIterator<kNdim>;
#endif
```

Reply:
Mm this would add a dependence of the `indexer.h` class to the `cuda/index_iterator` code